client/src/main/java/io/pravega/client/tables/TableSegment.java

```diff
@@ -0,0 +1,130 @@
+     *                returns a non-null value, this will perform an atomic Conditional Update where all the entries either
+     *                get applied or none will; otherwise a Blind Update will be performed.
+     * @return A CompletableFuture that, when completed, will contain a list of {@link KeyVersion}s associated with the
+     * newly inserted or updated entries. These will be in the same order as the entries given.
+     */
+    CompletableFuture<List<KeyVersion>> put(List<VersionedEntry<KeyT, ValueT>> entries);
+
+    /**
+     * Removes the given key from this Table Segment.
+     * @param key  The Key to remove. If {@link VersionedKey#getVersion()} is null, this will perform a Blind Update,
+     *             otherwise it will perform a Conditional Update based on the information provided.
+     * @return A CompletableFuture that, when completed, will indicate the Key has been removed.
+     */
+    CompletableFuture<Void> remove(VersionedKey<KeyT> key);
+
+    /**
+     * Removes one or more keys from this Table Segment.
+     *
+     * @param keys A list of keys to remove. If for at least one such key, {@link VersionedKey#getVersion()} returns a
+     *             non-null value, this will perform an atomic Conditional Update where all the keys either get removed
+     *             or none will; otherwise a Blind Update will be performed.
+     * @return A CompletableFuture that, when completed, will indicate that the keys have been removed.
+     */
+    CompletableFuture<Void> remove(List<VersionedKey<KeyT>> keys);
+
+    /**
+     * Gets the latest value for the given Key.
+     *
+     * @param key The Key to get the value for.
+     * @return A CompletableFuture that, when completed, will contain the requested result. If no such Key exists, this
+     * will be completed with a null value.
+     */
+    CompletableFuture<GetResult> get(KeyT key);
+
+    /**
+     * Gets the latest values for the given Keys.
+     *
+     * @param keys A list of Keys to get values for.
+     * @return A CompletableFuture that, when completed, will contain a list of results for the given Keys. The results
+     * will be in the same order as the keys. For any key that does not exist, a null value will be at its corresponding
+     * index.
+     */
+    CompletableFuture<List<GetResult<ValueT>>> get(List<KeyT> keys);
+
+    /**
+     * Registers a listener for all updates to a single key.
+     *
+     * @param listener The listener to register.
+     * @return A CompletableFuture that, when completed, will indicate the listener has been registered.
+     */
+    CompletableFuture<Void> registerListener(KeyUpdateListener<KeyT, ValueT> listener);
+
+    /**
+     * Unregisters a registered listener.
+     *
+     * @param listener THe listener to unregister.
+     */
+    void unregisterListener(KeyUpdateListener<KeyT, ValueT> listener);
```

Comment:
The listener does have a `close()` method. Right now they're both interfaces, so no implementation yet. I still need this `unregisterListener` method here since that's what the `close` would eventually invoke.

Reply:
I removed this method. Please see above: https://github.com/pravega/pravega/pull/2876#discussion_r213793807